Sources/Brave/Frontend/Shields/SubmitReportView.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  SwiftUIView.swift
+//  
+//
+//  Created by Jacob on 2023-09-21.
+//
+
+import SwiftUI
+import Strings
+import BraveUI
+import DesignSystem
+import BraveShields
+
+struct SubmitReportView: View {
+  let width: CGFloat
+  let url: URL
+  let dismiss: () -> Void
+  let submit: (String, String) -> Void
+  
+  @State var additionalDetails = ""
+  @State var contactDetails = ""
+  
+  private var borderShape: some InsettableShape {
+    RoundedRectangle(cornerRadius: 4, style: .continuous)
+  }
+  
+  private var textEditor: some View {
+    TextEditor(text: $additionalDetails)
+      .modifier(BaseBraveTextEditorStyleModifier(
+        placeholder: Strings.Shields.reportBrokenAdditionalDetails,
+        isPlaceholderVisible: additionalDetails.isEmpty
+      ))
+  }
+  
+  var body: some View {
+    ScrollView {
+      VStack(alignment: .leading, spacing: 16) {
+        Text(Strings.Shields.reportABrokenSite)
+          .font(.title)
+        Text(Strings.Shields.reportBrokenSiteBody1)
+        Text(url.absoluteString)
+          .foregroundStyle(Color(braveSystemName: .textInteractive))
+        Text(Strings.Shields.reportBrokenSiteBody2)
+          .font(.footnote)
+        
+        textEditor
+        
+        VStack(alignment: .leading, spacing: 4) {
+          Text(Strings.Shields.reportBrokenContactMe).font(.caption)
+          TextField(
+            Strings.Shields.reportBrokenContactMe,
+            text: $contactDetails, prompt: Text(Strings.Shields.reportBrokenContactMeSuggestions)
+          )
+          .textFieldStyle(BraveTextFieldStyle())
+          .autocorrectionDisabled()
+          .textInputAutocapitalization(.never)
+        }
+        
+        HStack {
+          Spacer()
+          Button(Strings.cancelButtonTitle, action: dismiss)
+          .buttonStyle(BraveOutlineButtonStyle(size: .large))
+          .multilineTextAlignment(.center)
+          
+          Button(Strings.Shields.reportBrokenSubmitButtonTitle, action: {
+            submit(additionalDetails, contactDetails)
+          })
+          .buttonStyle(BraveFilledButtonStyle(size: .large))
+          .multilineTextAlignment(.center)
+        }
+      }
+      .padding()
+      .foregroundStyle(Color(braveSystemName: .textSecondary))
+    }
+    .foregroundStyle(Color(braveSystemName: .textSecondary))
+    .frame(maxWidth: width)
+  }
+}
+
+#Preview {
+  SubmitReportView(
+    width: 300,
+    url: URL(string: "https://brave.com/privacy-features")!) {
+      // Do nothing
+    } submit: { _, _ in
+      // Do nothing
+    }
+}
```

Comment:
Needs to be either a legacy PreviewProvider or surrounded in conditional complication for Xcode 15 (`#if swift(>=5.9)`)